v3/src/models/data/formula-fn-registry.ts

```diff
@@ -15,13 +15,20 @@ const evaluateNode = (node: MathNode, scope?: FormulaMathJsScope) => {
 const cachedAggregateFnFactory =
 (fnName: string, fn: (args: MathNode[], mathjs: any, scope: FormulaMathJsScope) => FValue | FValue[]) => {
   return (args: MathNode[], mathjs: any, scope: FormulaMathJsScope) => {
-    const cacheKey = `${fnName}(${args.toString()})`
+    const cacheKey = `${fnName}(${args.toString()})-${scope.getCaseGroupId()}`
     const cachedValue = scope.getCached(cacheKey)
     if (cachedValue !== undefined) {
       return cachedValue
     }
     const result = fn(args, mathjs, scope)
-    scope.setCached(cacheKey, result)
+    // In practice, the cacheability is defined by the fact whether the function accesses attributes from the same
+    // collection only or not. It can be checked while the expression is calculated for the first time. `scope` object
```

Comment:
```suggestion
    // In practice, the cacheability is defined by whether or not the function accesses attributes from the same
    // collection only. It can be checked while the expression is calculated for the first time. `scope` object
```